### ISSUE TITLE
Login problems

### ISSUE THREAD

**Issue by TPBT-OFFICIAL** [einstein95/crunchy-xml-decoder#50]
I seem to be having problems login in, i have a free membership account that logins in fine but i have a premium (trial) account that doesn't want to login for some reason, i think theres been some recent changes on their website could anyone please help getting this solved?

**Comment by alzamer2** [comment]
Can you show us picture what you get when you log in 
You can datg and drop here

**Comment by TPBT-OFFICIAL** [comment]
You mean on crunchyroll or crunchy-xml-decoder?

**Comment by TPBT-OFFICIAL** [comment]
It seems it doesn't login if its a free trial but i canceled the free trial and tried again and it logged in (premium+ plan)

**Comment by alzamer2** [comment]
does the console close when you try to login or show that the login was failure, i want to know if the problem in the code or the structure of the login site

**Comment by TPBT-OFFICIAL** [comment]
Unfortunately I can't test that anymore but from what I remember after u tried to login it just said "login failed" and then closed.

**Comment by alzamer2** [comment]
Hmm
As you said i think the problem is at the site 
If i have free time at weekend i will try solve it but i cant promise any thing

**Comment by TPBT-OFFICIAL** [comment]
I can give you acess to a trial account I have if you want to test it.

**Comment by alzamer2** [comment]
No proplem i can create trial account
Thanks

**Comment by ganon11000** [comment]
I'm also having a login failed issue and after pressing any key the window instantly closes.

**Comment by draguelo** [comment]
I am having the same issue as ganon11000. I seem to recall it typing something in the password on it the last time I logged onto it. Now I'm not even sure if it's typing anything for the password or not but I keep trying to put in the password and it continually fails login and closes off just as it does for Ganon11000 in the picture he posted. Any assistance would be awesome. Thanks for the coding and hard work at getting this working

**Comment by ganon11000** [comment]
Out of interest does your password contain special characters?

**Comment by draguelo** [comment]
no, it doesn't contain any. I logged in just fine on the site and it logs in just fine on an older version of it. I would just use it if it'd actually download the videos like the newer version does. I've also tried uninstalling and reinstalling Python 2.7.5 as well as trying different versions of it an Pycrypto. I have a workaround for now so it's not like up the creek without a paddle. I can still use the guest login to download the subtitles so I download the video using a getflv downloader since that'll download the video without the subtitle. Maybe I can figure it out eventually but I'm going to retire for the evening and try again tomorrow sometime after work ends.

**Comment by danny-ml** [comment]
I am having the same issue as ganon11000. I can only login in the old version. I've tried it on two computers with different OS (Win 7, Win XP) with the same result.

http://i113.photobucket.com/albums/n213/danny-ml/Loging_Filed.png~original
http://i113.photobucket.com/albums/n213/danny-ml/Loging.png~original

**Comment by Davdomlan** [comment]
Same here... I am having the exact same problem,I cant loggin, any help would be great. Thank you

**Comment by albireon** [comment]
how can I get the old version?

**Comment by TPBT-OFFICIAL** [comment]
Like all of you, im also having the same problem AGAIN, it got fixed after i changed from free trial to actual premium account but just today i opened it and it didn't work again.

**Comment by ganon11000** [comment]
I got an old version to login and then edited the info from cookies.txt to the cookies file on the new version and all is now working

**Comment by TPBT-OFFICIAL** [comment]
@ganon11000 could you please share that version?

**Comment by alzamer2** [comment]
I'm sorry to say this but when i wrote the new login code i forget to add login test for trial so if your account is trial no mater what it will say it's failur so after the login consol close try to download

**Comment by ganon11000** [comment]
I'm not using the trial and get the issue still.
My work arround is (1.) using this https://userscloud.com/sh9nc1zl4cal (password is "github") on _make_cookies.bat -> (2.) attempting login on the new version -> (3.) opening cookies.txt from the old version and cookies from the new version and copying the old versions details listed here http://i.imgur.com/M5zpYYj.png to the corrosponding possisions here http://i.imgur.com/V9SFnxx.png (leaving the others as they were generated by the new version in step 2)

Hopefully this workaround will work for you and possibly help work out what's wrong?

**Comment by harmfuldata** [comment]
Was having the same issue. The cause appears to be in the check against the membership URL after the login attempt in the following file:
https://github.com/einstein95/crunchy-xml-decoder/blob/7162b201a845221b9f5325d2cff9056100efaa0b/crunchy-xml-decoder/login.py#L20

The current URL appears to generate a 404 error, and removing the trailing front slash appears to resolve the issue.

**Comment by TPBT-OFFICIAL** [comment]
@harmfuldata That fixed it, thank you
https://a.tpb.lc/xolgvc.png

**Comment by TPBT-OFFICIAL** [comment]
Confirmed fix from @harmfuldata 
You need to find the file login.py
Then edit the following line:
ite = session.get('https://www.crunchyroll.com/acct/membership/', headers=headers, verify=True).text
You need to change it to
ite = session.get('https://www.crunchyroll.com/acct/membership', headers=headers, verify=True).text

In the end it should look like something like this:
https://a.tpb.lc/cqtlkx.png

After that try to login again and it should be working fine.

**Comment by danny-ml** [comment]
Confirmed fix from @harmfuldata
Thank you very much.
![login_successful](https://cloud.githubusercontent.com/assets/15720798/12226879/a89f9dd8-b7e8-11e5-85be-351482aad750.png)

**Comment by Davdomlan** [comment]
Thanks @harmfuldata, I worked like a charm.

**Comment by CraigMack69** [comment]
thanks for the fix

**Comment by R9delta** [comment]
Editing the login.py didn't seem to do the trick for me so I personally found a way around it. 
Your cookie file (just "cookie", no extension) which should be in whatever same folder you've got crunchy-xml-decoder.py extracted in should look like so:
![crunchyrollcookieexample](https://cloud.githubusercontent.com/assets/17975780/13911713/12ac0446-ef02-11e5-85e4-7d04daa16c7f.png)

...with the exception of your own proper values filled in the blank spaces. Obviously mine are edited out. You can find out the values you need by using an internet browser like Google Chrome and having the extension "EditThisCookie" Installed. Log in the crunchyroll.com, right click anywhere on the page and click Edit this Cookie. You'll find everything you need so edit them in and save.
